clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -101,6 +101,17 @@
 
     private int assignmentId = 0;
 
+    @Override
+    public String toString() {
```

Comment:
nit: The first `toString()` you added made more sense in the context of the new log message. It just didn't handle the case of manual assignment. I think the warning will be a little more consumable if we just mention the state that is relevant to it. Could we have a `prettyString()` or something which gives a nice representation of the user subscription/assignment?

Reply:
I agree, would change it again.

Reply:
I can't tell if you're trolling me 😜 . The method `prettyString` below still does not take handle manual assignment. Maybe it could be something like this?
```java
public String prettyString() {
  switch (subscriptionType) {
    case NONE: 
      return "None";
    case AUTO_TOPICS:
      return "subscribe($topics)";
    case AUTO_PATTERN:
      return "subscribe($pattern)";
    case USER_ASSIGNED:
      return "assign($partitions)";
  }
}
```

Reply:
Ah I missed it --- so no I was not intentional but just need more coffee :P Will change again.